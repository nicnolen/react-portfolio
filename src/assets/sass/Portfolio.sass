@import './variables'
@import './index'
@import './App'

.card
  max-width: 100%

.img-container
  max-width: 100%
  max-height: 100%
  
.project
  max-width:100%
  max-height:100%

.content 
  background: $fadeImage, url(#{$portfolioImagePath}/project-background.jpg)
  background-size: cover
  background-position: center
  display: flex
  justify-content: space-between
  align-items: center
  padding-right: 2vw
  color: $gold
  font-size: 4vw
 

.github-link 
  background-color: $gold

.card-title
  display: inline-block
  color: $dark-green

