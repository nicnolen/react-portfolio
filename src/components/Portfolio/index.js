import React from 'react';
import projectInfo from './projects.json';
import Project from '../Project';
import 'bootstrap/dist/css/bootstrap.min.css';

const Wrapper = props => {
  return <div className='wrapper'>{props.children}</div>;
};

function Portfolio() {
  return (
    <section className='container'>
      <div className='project'>
        <h2 className='top-title'>Professional Portfolio</h2>
        <hr></hr>
      </div>

      <Wrapper id='card-data'>
        {projectInfo.map(project => (
          <Project
            key={project.id}
            title={project.title}
            image={project.image}
            description={project.description}
            github={project.github}
            deployed={project.deployed}
            tools={project.tools}
          />
        ))}
      </Wrapper>
    </section>
  );
}

export default Portfolio;
