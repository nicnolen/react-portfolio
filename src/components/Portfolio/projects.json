[
  {
    "id": "1",
    "title": "Hawker",
    "description": "A marketplace for buying and selling items.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/hawker-home.png",
    "github": "https://github.com/nicnolen/Hawker",
    "deployed": "https://texas-hawker.herokuapp.com/",
    "tools": "JavaScript/Node.js/Inquirer/OOP"
  },
  {
    "id": "2",
    "title": "Travel Lovers",
    "description": "Professional portfolio to showcase skills to employers.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/travel-lovers.png",
    "github": "https://github.com/nicnolen/Travel-Lovers",
    "deployed": "https://travel--lovers.herokuapp.com/",
    "tools": "Handlebars.js/Sequelize/SASS/bcrypt"
  },
  {
    "id": "3",
    "title": "Watchopedia",
    "description": "A website that allows a user to easily search for a movie, TV show, or actor information through API requests.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/watchopedia-homepage.png",
    "github": "https://github.com/nicnolen/watchopedia",
    "deployed": "https://nicnolen.github.io/watchopedia/",
    "tools": "HTML/CSS/Bulma/JavaScript"
  },
  {
    "id": "4",
    "title": "Just Tech News",
    "description": "This app allows a user to post an article from another source and other users can vote or comment on the post.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/just-tech-news-dashboard.png",
    "github": "https://github.com/nicnolen/tech-news-java-api",
    "deployed": "https://cc-java-api-32.herokuapp.com/",
    "tools": "Java/Spring Boot/Thymeleaf/Spring Session/MySQL"
  },
  {
    "id": "5",
    "title": "Shop Shop",
    "description": "E-commerce application that sells some of the finest items available on the internet.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/shop-shop.png",
    "github": "https://github.com/nicnolen/shop-shop",
    "deployed": "https://dry-shelf-71701.herokuapp.com/",
    "tools": "MongoDB/Express.js/React.js/Node.js/TDD"
  },
  {
    "id": "6",
    "title": "Weather Dashboard",
    "description": "A weather dashboard that dynamically displays the current weather and five day forecast for searched cities.",
    "image": "http://nicnolen.github.io/react-portfolio/assets/images/projectImages/weather-homepage.png",
    "github": "https://github.com/nicnolen/weather-dashboard",
    "deployed": "https://nicnolen.github.io/weather-dashboard/",
    "tools": "HTML/CSS/Bootstrap/JavaScript/moments.js"
  }
]
