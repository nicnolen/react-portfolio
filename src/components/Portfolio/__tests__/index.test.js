// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Portfolio component', () => {
  // verify the component is rendering properly
  it('Portfolio component renders', () => {
    render(<Portfolio />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Portfolio />);
    expect(asFragment()).toMatchSnapshot();
  });
});