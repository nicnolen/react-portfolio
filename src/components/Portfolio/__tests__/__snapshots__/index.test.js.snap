// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <div
      class="project"
    >
      <h2
        class="top-title"
      >
        Professional Portfolio
      </h2>
      <hr />
    </div>
    <div
      class="wrapper"
    >
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          A marketplace for buying and selling items.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://texas-hawker.herokuapp.com/"
          >
            <img
              alt="A marketplace for buying and selling items."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/hawker-home.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/Hawker"
              title="A marketplace for buying and selling items."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Hawker
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            JavaScript/Node.js/Inquirer/OOP
          </h4>
        </div>
      </div>
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          Professional portfolio to showcase skills to employers.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://travel--lovers.herokuapp.com/"
          >
            <img
              alt="Professional portfolio to showcase skills to employers."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/travel-lovers.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/Travel-Lovers"
              title="Professional portfolio to showcase skills to employers."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Travel Lovers
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            Handlebars.js, Sequelize, SASS, bcrypt
          </h4>
        </div>
      </div>
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          A website that allows a user to easily search for a movie, TV show, or actor information through API requests.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://nicnolen.github.io/watchopedia/"
          >
            <img
              alt="A website that allows a user to easily search for a movie, TV show, or actor information through API requests."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/watchopedia-homepage.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/watchopedia"
              title="A website that allows a user to easily search for a movie, TV show, or actor information through API requests."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Watchopedia
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            HTML/CSS/Bulma/JavaScript
          </h4>
        </div>
      </div>
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          E-commerce application that sells some of the finest items available on the internet.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://dry-shelf-71701.herokuapp.com/"
          >
            <img
              alt="E-commerce application that sells some of the finest items available on the internet."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/shop-shop.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/shop-shop"
              title="E-commerce application that sells some of the finest items available on the internet."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Shop Shop
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            MongoDB, Express.js, React.js, Node.js, TDD
          </h4>
        </div>
      </div>
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          This app allows a user to post an article from another source and other users can vote or comment on the post.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://just--tech--news.herokuapp.com/"
          >
            <img
              alt="This app allows a user to post an article from another source and other users can vote or comment on the post."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/just-tech-news-dashboard.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/just-tech-news"
              title="This app allows a user to post an article from another source and other users can vote or comment on the post."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Just Tech News
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            MySQL/Handlebars.js/ORM/Express.js
          </h4>
        </div>
      </div>
      <div
        class="card mb-5"
      >
        <div
          class="description"
        >
          A weather dashboard that dynamically displays the current weather and five day forecast for searched cities.
        </div>
        <figure
          class="img-container mb-0"
        >
          <a
            href="https://nicnolen.github.io/weather-dashboard/"
          >
            <img
              alt="A weather dashboard that dynamically displays the current weather and five day forecast for searched cities."
              class="project"
              src="http://nicnolen.github.io/react-portfolio/assets/images/projectImages/weather-homepage.png"
            />
          </a>
        </figure>
        <div
          class="content d-flex justify-content-between align-items-center"
        >
          <span
            class="github-link"
          >
            <a
              href="https://github.com/nicnolen/weather-dashboard"
              title="A weather dashboard that dynamically displays the current weather and five day forecast for searched cities."
            >
              <img
                alt="GitHub Repo"
                src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
              />
              <h3
                class="card-title"
              >
                Weather Dashboard
              </h3>
            </a>
          </span>
          <h4
            class="project-skills fs-4 pb-1 px-1"
          >
            HTML/CSS/Bootstrap/JavaScript/moments.js
          </h4>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
